### PR TITLE
(FACT-1616) update the OS specific facts

### PR DESCRIPTION
**Pull request by ericldelaney**
Updated the OS specific fact tests with Code Style, TestRail ID's, and risk tags
This update is all do..end blocks, indentation, comments and test_name strings

NOTE: I did NOT do any updates for supported platforms that where added or removed

### CODE REVIEW

**Comment by ericldelaney** [comment]
Tested on:
arista4-32a, ciscoxr-64a%7Bhypervisor=vmpooler%7D, cisconx-64a%7Bhypervisor=vmpooler%7D, cumulus25-64a, debian8-64a, fedora25-64a, osx1012-64a, redhat7-64a, sles12-64a, solaris11-64a, ubuntu1610-64a, windows10ent-64a, windows2016-64a

Still need to test on AIX

**Comment by ericldelaney** [comment]
@johnduarte I had thought the same thing, but I didn't want to make all the changes at one time to make it easier to review.

I plan to update the supported OS versions and look at converting all the on_fact() calls into just a call or 2 to facter --yaml and then parse the output since it shouldn't be changing between calls.

File: acceptance/tests/facts/fedora.rb [hunk 1]
```diff
@@ -1,80 +1,90 @@
...
-    when '22' then '4.0'
-    else 'undefined'
-  end
+    step "Ensure the kernel fact resolves as expected" do
+      kernel_version = case os_version
+                         when '20' then
+                           '3.11'
+                         when '21' then
+                           '3.17'
+                         when '22' then
```
Comment (joshcooper) [inline acceptance/tests/facts/fedora.rb]: `then` is only needed in single-line `when` statements, so it's not needed when you break it into multiple lines. It doesn't hurt anything to have them, but for future reference.

**Comment by joshcooper** [comment]
Ignoring whitespace changes:

```diff
$ git show -w upstream/pr/1563
commit 75989b44e9217809bce58f9efa73078c039ff751
Author: Eric Delaney <eric.delaney@puppet.com>
Date:   Wed Apr 26 13:40:15 2017 -0700

    (FACT-1616) update the OS specific facts
    
    Update the OS specific fact tests with Code Style, TestRail ID's, and risk tags
    Updated a few comments and test names to be more generic
    
    NOTE: I did NOT do any updates for supported platforms that where added or removed

diff --git a/acceptance/tests/facts/aix.rb b/acceptance/tests/facts/aix.rb
index 1de0b1d..5e6e951 100644
--- a/acceptance/tests/facts/aix.rb
+++ b/acceptance/tests/facts/aix.rb
@@ -1,8 +1,9 @@
-test_name "Facts should resolve as expected in AIX 5.3, 6.1, 7.1 and 7.2"
+test_name "C69636: Facts should resolve as expected on AIX" do
+  tag 'risk:high'
 
 #
 # This test is intended to ensure that facts specific to an OS configuration
-# resolve as expected in AIX 5.3, 6.1, and 7.1.
+# resolve as expected on AIX.
 #
 # Facts tested: os, processors, networking, identity, kernel
 #
@@ -28,8 +29,7 @@ agents.each do |agent|
     os_arch     = /[Pp]ower[Pp][Cc]/
     os_hardware = /IBM/
 
-  step "Ensure the OS fact resolves as expected"
-
+    step "Ensure the OS fact resolves as expected" do
       expected_os = {
           'os.architecture'  => os_arch,
           'os.family'        => 'AIX',
@@ -42,9 +42,9 @@ agents.each do |agent|
       expected_os.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the Processors fact resolves with reasonable values"
-
+    step "Ensure the Processors fact resolves with reasonable values" do
       expected_processors = {
           'processors.count'  => /[1-9]+/,
           'processors.isa'    => os_arch,
@@ -54,8 +54,9 @@ agents.each do |agent|
       expected_processors.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the identity fact resolves as expected"
+    step "Ensure the identity fact resolves as expected" do
       expected_identity = {
           'identity.gid'        => '0',
           'identity.group'      => 'system',
@@ -67,8 +68,9 @@ agents.each do |agent|
       expected_identity.each do |fact, value|
         assert_equal(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the kernel fact resolves as expected"
+    step "Ensure the kernel fact resolves as expected" do
       expected_kernel = {
           'kernel'           => 'AIX',
           'kernelrelease'    => kernel_release,
@@ -80,3 +82,5 @@ agents.each do |agent|
         assert_match(value, fact_on(agent, fact))
       end
     end
+  end
+end
diff --git a/acceptance/tests/facts/debian.rb b/acceptance/tests/facts/debian.rb
index c3583f7..eba78b3 100644
--- a/acceptance/tests/facts/debian.rb
+++ b/acceptance/tests/facts/debian.rb
@@ -1,8 +1,9 @@
-test_name "Facts should resolve as expected in Debian 6 and 7"
+test_name "Facts should resolve as expected in Debian" do
+  tag 'risk:high'
 
 #
 # This test is intended to ensure that facts specific to an OS configuration
-# resolve as expected in Debian 6, 7 and 8.
+# resolve as expected in Debian.
 #
 # Facts tested: os, processors, networking, identity, kernel
 #
@@ -32,7 +33,7 @@ agents.each do |agent|
       os_hardware = 'i686'
     end
 
-  step "Ensure the OS fact resolves as expected"
+    step "Ensure the OS fact resolves as expected" do
       expected_os = {
           'os.architecture'         => os_arch,
           'os.distro.codename'      => codename,
@@ -52,8 +53,9 @@ agents.each do |agent|
       expected_os.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the Processors fact resolves with reasonable values"
+    step "Ensure the Processors fact resolves with reasonable values" do
       expected_processors = {
           'processors.count'         => /[1-9]/,
           'processors.physicalcount' => /[1-9]/,
@@ -64,8 +66,9 @@ agents.each do |agent|
       expected_processors.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the identity fact resolves as expected"
+    step "Ensure the identity fact resolves as expected" do
       expected_identity = {
           'identity.gid'        => '0',
           'identity.group'      => 'root',
@@ -77,8 +80,9 @@ agents.each do |agent|
       expected_identity.each do |fact, value|
         assert_equal(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the kernel fact resolves as expected"
+    step "Ensure the kernel fact resolves as expected" do
       expected_kernel = {
           'kernel'           => 'Linux',
           'kernelrelease'    => os_kernel,
@@ -90,3 +94,5 @@ agents.each do |agent|
         assert_match(value, fact_on(agent, fact))
       end
     end
+  end
+end
diff --git a/acceptance/tests/facts/el.rb b/acceptance/tests/facts/el.rb
index c026979..3faa036 100644
--- a/acceptance/tests/facts/el.rb
+++ b/acceptance/tests/facts/el.rb
@@ -1,8 +1,9 @@
-test_name "Facts should resolve as expected in EL 4, 5, 6 and 7"
+test_name "Facts should resolve as expected on EL" do
+  tag 'risk:high'
 
 #
 # This test is intended to ensure that facts specific to an OS configuration
-# resolve as expected in EL 4, 5, 6 and 7.
+# resolve as expected on EL.
 #
 # Facts tested: os, processors, networking, identity, kernel
 #
@@ -53,8 +54,7 @@ agents.each do |agent|
       processor_model_pattern = 'Intel\(R\)'
     end
 
-  step "Ensure the OS fact resolves as expected"
-
+    step "Ensure the OS fact resolves as expected" do
       expected_os = {
           'os.architecture'  => os_arch,
           'os.family'        => 'RedHat',
@@ -68,9 +68,9 @@ agents.each do |agent|
       expected_os.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the Processors fact resolves with reasonable values"
-
+    step "Ensure the Processors fact resolves with reasonable values" do
       expected_processors = {
           'processors.count'         => /[1-9]/,
           'processors.physicalcount' => /[1-9]/,
@@ -81,8 +81,9 @@ agents.each do |agent|
       expected_processors.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the identity fact resolves as expected"
+    step "Ensure the identity fact resolves as expected" do
       expected_identity = {
           'identity.gid'        => '0',
           'identity.group'      => 'root',
@@ -94,9 +95,9 @@ agents.each do |agent|
       expected_identity.each do |fact, value|
         assert_equal(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the kernel fact resolves as expected"
-
+    step "Ensure the kernel fact resolves as expected" do
       expected_kernel = {
           'kernel'           => 'Linux',
           'kernelrelease'    => kernel_version,
@@ -108,3 +109,5 @@ agents.each do |agent|
         assert_match(value, fact_on(agent, fact))
       end
     end
+  end
+end
diff --git a/acceptance/tests/facts/fedora.rb b/acceptance/tests/facts/fedora.rb
index 4d05be9..5bd88e7 100644
--- a/acceptance/tests/facts/fedora.rb
+++ b/acceptance/tests/facts/fedora.rb
@@ -1,8 +1,9 @@
-test_name "Facts should resolve as expected in Fedora 20, 21, and 22"
+test_name "Facts should resolve as expected on Fedora" do
+  tag 'risk:high'
 
 #
 # This test is intended to ensure that facts specific to an OS configuration
-# resolve as expected in Fedora 20, 21, and 22.
+# resolve as expected on Fedora.
 #
 # Facts tested: os, processors, networking, identity, kernel
 #
@@ -20,7 +21,7 @@ agents.each do |agent|
       os_hardware = 'i686'
     end
 
-  step "Ensure the OS fact resolves as expected"
+    step "Ensure the OS fact resolves as expected" do
       expected_os = {
           'os.architecture'  => os_arch,
           'os.family'        => 'RedHat',
@@ -33,8 +34,9 @@ agents.each do |agent|
       expected_os.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the Processors fact resolves with reasonable values"
+    step "Ensure the Processors fact resolves with reasonable values" do
       expected_processors = {
           'processors.count'         => /[1-9]/,
           'processors.physicalcount' => /[1-9]/,
@@ -45,8 +47,9 @@ agents.each do |agent|
       expected_processors.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the identity fact resolves as expected"
+    step "Ensure the identity fact resolves as expected" do
       expected_identity = {
           'identity.gid'        => '0',
           'identity.group'      => 'root',
@@ -58,13 +61,18 @@ agents.each do |agent|
       expected_identity.each do |fact, value|
         assert_equal(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the kernel fact resolves as expected"
+    step "Ensure the kernel fact resolves as expected" do
       kernel_version = case os_version
-    when '20' then '3.11'
-    when '21' then '3.17'
-    when '22' then '4.0'
-    else 'undefined'
+                         when '20' then
+                           '3.11'
+                         when '21' then
+                           '3.17'
+                         when '22' then
+                           '4.0'
+                         else
+                           'undefined'
                        end
 
       expected_kernel = {
@@ -78,3 +86,5 @@ agents.each do |agent|
         assert_match(value, fact_on(agent, fact))
       end
     end
+  end
+end
diff --git a/acceptance/tests/facts/macosx.rb b/acceptance/tests/facts/macosx.rb
index 54bc01e..e1dd457 100644
--- a/acceptance/tests/facts/macosx.rb
+++ b/acceptance/tests/facts/macosx.rb
@@ -1,8 +1,9 @@
-test_name "Facts should resolve as expected in Mac OS X 10.9 and 10.10"
+test_name "Facts should resolve as expected on Mac OS X" do
+  tag 'risk:high'
 
 #
 # This test is intended to ensure that facts specific to an OS configuration
-# resolve as expected in Mac OS X 10.9 and 10.10
+# resolve as expected on Mac OS X
 #
 # Facts tested: os, processors, networking, identity, kernel
 #
@@ -27,8 +28,7 @@ agents.each do |agent|
         fail_test("Unknown os_version #{os_version}")
     end
 
-  step "Ensure the OS fact resolves as expected"
-
+    step "Ensure the OS fact resolves as expected" do
       expected_os = {
           'os.architecture'         => 'x86_64',
           'os.family'               => 'Darwin',
@@ -47,9 +47,9 @@ agents.each do |agent|
       expected_os.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the Processors fact resolves with reasonable values"
-
+    step "Ensure the Processors fact resolves with reasonable values" do
       expected_processors = {
           'processors.count'         => /[1-9]/,
           'processors.physicalcount' => /[1-9]/,
@@ -60,8 +60,9 @@ agents.each do |agent|
       expected_processors.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the identity fact resolves as expected"
+    step "Ensure the identity fact resolves as expected" do
       expected_identity = {
           'identity.gid'        => '0',
           'identity.group'      => 'wheel',
@@ -73,9 +74,9 @@ agents.each do |agent|
       expected_identity.each do |fact, value|
         assert_equal(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the kernel fact resolves as expected"
-
+    step "Ensure the kernel fact resolves as expected" do
       expected_kernel = {
           'kernel'           => 'Darwin',
           'kernelrelease'    => /#{kernel_major}\.\d\.\d/,
@@ -87,3 +88,5 @@ agents.each do |agent|
         assert_match(value, fact_on(agent, fact))
       end
     end
+  end
+end
\ No newline at end of file
diff --git a/acceptance/tests/facts/networking_facts.rb b/acceptance/tests/facts/networking_facts.rb
index da7dab4..6bddeb3 100644
--- a/acceptance/tests/facts/networking_facts.rb
+++ b/acceptance/tests/facts/networking_facts.rb
@@ -1,4 +1,5 @@
-test_name 'FACT-1361 - C59029 networking facts should be fully populated' do
+test_name 'C59029: networking facts should be fully populated' do
+  tag 'risk:high'
 
 #
 # This test is intended to ensure that networking facts resolves
diff --git a/acceptance/tests/facts/ruby.rb b/acceptance/tests/facts/ruby.rb
index 3d0ec6a..c1c3c6c 100644
--- a/acceptance/tests/facts/ruby.rb
+++ b/acceptance/tests/facts/ruby.rb
@@ -1,4 +1,5 @@
-test_name "The Ruby fact should resolve as expected in AIO"
+test_name "The Ruby fact should resolve as expected in AIO" do
+  tag 'risk:high'
 
 #
 # This test is intended to ensure that the the ruby fact resolves
@@ -7,7 +8,7 @@ test_name "The Ruby fact should resolve as expected in AIO"
 
   skip_test "Ruby fact test is confined to AIO" if @options[:type] != 'aio'
 
-step "Ensure the Ruby fact resolves as expected"
+  step "Ensure the Ruby fact resolves as expected" do
 
     case agent['platform']
       when /windows/
@@ -44,3 +45,5 @@ expected_ruby = {
     expected_ruby.each do |fact, value|
       assert_match(value, fact_on(agent, fact))
     end
+  end
+end
diff --git a/acceptance/tests/facts/sles.rb b/acceptance/tests/facts/sles.rb
index 278f758..e0a1456 100644
--- a/acceptance/tests/facts/sles.rb
+++ b/acceptance/tests/facts/sles.rb
@@ -1,8 +1,9 @@
-test_name "Facts should resolve as expected in SLES 10, 11 and 12"
+test_name "Facts should resolve as expected on SLES" do
+  tag 'risk:high'
 
 #
 # This test is intended to ensure that facts specific to an OS configuration
-# resolve as expected in SLES 10, 11 and 12.
+# resolve as expected on SLES.
 #
 # Facts tested: os, processors, networking, identity, kernel
 #
@@ -32,8 +33,7 @@ agents.each do |agent|
       processor_model_pattern = 'Intel\(R\)'
     end
 
-  step "Ensure the OS fact resolves as expected"
-
+    step "Ensure the OS fact resolves as expected" do
       expected_os = {
           'os.architecture'  => os_arch,
           'os.family'        => 'Suse',
@@ -47,9 +47,9 @@ agents.each do |agent|
       expected_os.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the Processors fact resolves with reasonable values"
-
+    step "Ensure the Processors fact resolves with reasonable values" do
       expected_processors = {
           'processors.count'         => /[1-9]/,
           'processors.physicalcount' => /[1-9]/,
@@ -60,8 +60,9 @@ agents.each do |agent|
       expected_processors.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the identity fact resolves as expected"
+    step "Ensure the identity fact resolves as expected" do
       expected_identity = {
           'identity.gid'        => '0',
           'identity.group'      => 'root',
@@ -73,8 +74,9 @@ agents.each do |agent|
       expected_identity.each do |fact, value|
         assert_equal(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the kernel fact resolves as expected"
+    step "Ensure the kernel fact resolves as expected" do
       if os_version == '10'
         kernel_version = '2.6'
       elsif os_version == '11'
@@ -94,3 +96,5 @@ agents.each do |agent|
         assert_match(value, fact_on(agent, fact))
       end
     end
+  end
+end
diff --git a/acceptance/tests/facts/solaris.rb b/acceptance/tests/facts/solaris.rb
index 45f1d1b..d418308 100644
--- a/acceptance/tests/facts/solaris.rb
+++ b/acceptance/tests/facts/solaris.rb
@@ -1,8 +1,9 @@
-test_name "Facts should resolve as expected in Solaris 10 and 11"
+test_name "C15496: Facts should resolve as expected on Solaris" do
+  tag 'risk:high'
 
 #
 # This test is intended to ensure that facts specific to an OS configuration
-# resolve as expected in Solaris 10 and 11.
+# resolve as expected on Solaris
 #
 # Facts tested: os, processors, networking, identity, kernel
 #
@@ -34,7 +35,7 @@ agents.each do |agent|
         proc_isa        = /i386/
     end
 
-  step "Ensure the OS fact resolves as expected"
+    step "Ensure the OS fact resolves as expected" do
       expected_os = {
           'os.architecture'  => os_architecture,
           'os.family'        => 'Solaris',
@@ -48,8 +49,9 @@ agents.each do |agent|
       expected_os.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the Processors fact resolves with reasonable values"
+    step "Ensure the Processors fact resolves with reasonable values" do
       expected_processors = {
           'processors.count'         => /[1-9]/,
           'processors.physicalcount' => /[1-9]/,
@@ -60,8 +62,9 @@ agents.each do |agent|
       expected_processors.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the identity fact resolves as expected"
+    step "Ensure the identity fact resolves as expected" do
       expected_identity = {
           'identity.gid'        => '0',
           'identity.group'      => 'root',
@@ -73,8 +76,9 @@ agents.each do |agent|
       expected_identity.each do |fact, value|
         assert_equal(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the kernel fact resolves as expected"
+    step "Ensure the kernel fact resolves as expected" do
       expected_kernel = {
           'kernel'           => 'SunOS',
           'kernelrelease'    => "5.#{os_version}",
@@ -86,3 +90,5 @@ agents.each do |agent|
         assert_match(value, fact_on(agent, fact))
       end
     end
+  end
+end
diff --git a/acceptance/tests/facts/ubuntu.rb b/acceptance/tests/facts/ubuntu.rb
index e7342c1..fe19574 100644
--- a/acceptance/tests/facts/ubuntu.rb
+++ b/acceptance/tests/facts/ubuntu.rb
@@ -1,4 +1,5 @@
-test_name "Facts should resolve as expected in Ubuntu"
+test_name "C15288: Facts should resolve as expected on Ubuntu" do
+  tag 'risk:high'
 
 #
 # This test is intended to ensure that facts specific to an OS configuration
@@ -54,7 +55,7 @@ agents.each do |agent|
       os_hardware = 'i686'
     end
 
-  step "Ensure the OS fact resolves as expected"
+    step "Ensure the OS fact resolves as expected" do
       expected_os = {
           'os.architecture'         => os_arch,
           'os.distro.codename'      => os_name,
@@ -72,8 +73,9 @@ agents.each do |agent|
       expected_os.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the Processors fact resolves with reasonable values"
+    step "Ensure the Processors fact resolves with reasonable values" do
       expected_processors = {
           'processors.count'         => /[1-9]/,
           'processors.physicalcount' => /[1-9]/,
@@ -84,8 +86,9 @@ agents.each do |agent|
       expected_processors.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the identity fact resolves as expected"
+    step "Ensure the identity fact resolves as expected" do
       expected_identity = {
           'identity.gid'        => '0',
           'identity.group'      => 'root',
@@ -97,8 +100,9 @@ agents.each do |agent|
       expected_identity.each do |fact, value|
         assert_equal(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the kernel fact resolves as expected"
+    step "Ensure the kernel fact resolves as expected" do
       expected_kernel = {
           'kernel'           => 'Linux',
           'kernelrelease'    => os_kernel,
@@ -110,3 +114,5 @@ agents.each do |agent|
         assert_match(value, fact_on(agent, fact))
       end
     end
+  end
+end
diff --git a/acceptance/tests/facts/windows.rb b/acceptance/tests/facts/windows.rb
index 2a8c1fd..20e138b 100644
--- a/acceptance/tests/facts/windows.rb
+++ b/acceptance/tests/facts/windows.rb
@@ -1,4 +1,5 @@
-test_name "Facts should resolve as expected on Windows platforms"
+test_name "C26851: Facts should resolve as expected on Windows platforms" do
+  tag 'risk:high'
 
 #
 # This test is intended to ensure that facts specific to an OS configuration
@@ -32,8 +33,7 @@ agents.each do |agent|
     end
 
 
-  step "Ensure the OS fact resolves as expected"
-
+    step "Ensure the OS fact resolves as expected" do
       expected_os = {
           'os.architecture'     => agent['platform'] =~ /64/ ? 'x64' : 'x86',
           'os.family'           => 'windows',
@@ -47,9 +47,9 @@ agents.each do |agent|
       expected_os.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the Processors fact resolves with reasonable values"
-
+    step "Ensure the Processors fact resolves with reasonable values" do
       expected_processors = {
           'processors.count'         => /[1-9]/,
           'processors.physicalcount' => /[1-9]/,
@@ -60,8 +60,9 @@ agents.each do |agent|
       expected_processors.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the identity fact resolves as expected"
+    step "Ensure the identity fact resolves as expected" do
       # Regular expression to validate the username from facter in the form of '<domain>\<username>'
       # Reference - https://msdn.microsoft.com/en-us/library/bb726984.aspx
       # - The domain name can be any character or empty
@@ -75,9 +76,9 @@ agents.each do |agent|
       expected_identity.each do |fact, value|
         assert_match(value, fact_on(agent, fact))
       end
+    end
 
-  step "Ensure the kernel fact resolves as expected"
-
+    step "Ensure the kernel fact resolves as expected" do
       expected_kernel = {
           'kernel'           => 'windows',
           'kernelrelease'    => kernel_version,
@@ -89,3 +90,5 @@ agents.each do |agent|
         assert_match(value, fact_on(agent, fact))
       end
     end
+  end
+end
```

**Pull request merged**